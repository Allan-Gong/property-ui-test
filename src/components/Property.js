import React from "react";
import PropTypes from "prop-types";
import { Card, Image, Button } from "semantic-ui-react";

const Property = ({
  property,
  buttonText,
  buttonAction,
  buttonDisabled = false
}) => {
  const headerStyle = {
    background: property.agency.brandingColors.primary
  };

  return (
    <Card fluid className="property" key={property.id}>
      <Card.Header className="property-header" style={headerStyle}>
        <img
          src={property.agency.logo}
          className="agency-logo"
          alt="Agency logo"
        />
      </Card.Header>
      <Image src={property.mainImage} className="main-image" alt="Main" />
      <Card.Content>
        <p className="left floated price">{property.price}</p>
        <Button
          primary
          disabled={buttonDisabled}
          className="right floated button-action"
          onClick={() => buttonAction(property.id)}
        >
          {buttonText}
        </Button>
      </Card.Content>
    </Card>
  );
};

Property.propTypes = {
  property: PropTypes.shape({
    price: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    mainImage: PropTypes.string.isRequired,
    agency: PropTypes.shape({
      brandingColors: PropTypes.shape({
        primary: PropTypes.string.isRequired
      }).isRequired,
      logo: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  buttonText: PropTypes.string.isRequired,
  buttonAction: PropTypes.func.isRequired,
  buttonDisabled: PropTypes.bool.isRequired
};

export default Property;
