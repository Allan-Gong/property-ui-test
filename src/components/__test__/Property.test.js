import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Property from '../Property';

const TEST_PROPERTY_DATA = {
  "price": "$726,500",
  "agency": {
    "brandingColors": {
      "primary": "#ffe512"
    },
    "logo": "http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
  },
  "id": "1",
  "mainImage": "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
};

describe('Property', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Property property={TEST_PROPERTY_DATA} buttonText="text" buttonAction={() => {}} />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should invoke given buttonAction from props as expected', () => {
    const buttonAction = sinon.spy();

    const output = shallow(
      <Property property={TEST_PROPERTY_DATA} buttonText="text" buttonAction={buttonAction} />
    );
    // expect(wrapper.find(Link).length).toBe(5);
    output.find('.button-action').simulate('click');
    expect(buttonAction.calledOnce);//.to.have.property('callCount', 1);
  });
});

