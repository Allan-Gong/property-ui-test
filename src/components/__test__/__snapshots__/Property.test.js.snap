// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property should render correctly 1`] = `
<Card
  className="property"
  fluid={true}
  key="1"
>
  <CardHeader
    className="property-header"
    style={
      Object {
        "background": "#ffe512",
      }
    }
  >
    <img
      alt="Agency logo"
      className="agency-logo"
      src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
    />
  </CardHeader>
  <Image
    alt="Main"
    as="img"
    className="main-image"
    src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
    ui={true}
  />
  <CardContent>
    <p
      className="left floated price"
    >
      $726,500
    </p>
    <Button
      as="button"
      className="right floated button-action"
      onClick={[Function]}
      primary={true}
    >
      text
    </Button>
  </CardContent>
</Card>
`;
